blog/2025/issue-credentials-over-openid4vci.adoc

```diff
@@ -0,0 +1,356 @@
+
+Keycloak implements OpenID4VCI, enabling the issuance of verifiable credentials (VCs) as digital proofs of identity or attributes. Configuring this functionality requires consistent setup across the realm, clients, and issuable credentials (client scopes).
+
+For example, consider the following scenario: The Keycloak OAuth SIG team wants to issue verifiable membership credentials to its members, including their name and email, which can later be presented at onsite or virtual events as proof of active membership.
+
+NOTE: At the time of writing this blog, Keycloak's support for OpenID4VCI is still experimental. This guide uses the https://www.keycloak.org/nightly/[**nightly release**] which builds from the main branch. The feature is expected to be promoted to preview in early 2026.
+
+== Introduction to OpenID4VCI
+
+OpenID4VCI is a protocol developed by the OpenID Foundation that extends the OAuth 2.0 framework to support the secure and interoperable issuance of Verifiable Credentials (VCs). VCs are digital, tamper-evident representations of information, such as identity attributes or qualifications, that can be cryptographically verified without needing to contact the issuer at verification time. This enables decentralized identity ecosystems in which users (holders) retain control of their data and can share it selectively with verifiers.
+
+.Triangle of trust or Issuer-Holder-Verifier model
+image::${blogImages}/openid4vci/triangle-of-trust.png[alt="Triangle of trust or Issuer-Holder-Verifier model", align=center, width=800]
+
+=== Why Use OpenID4VCI
+
+The primary motivations for adopting OpenID4VCI include:
+
+- **Interoperability**: Builds on established OIDC standards, simplifying integration with existing identity providers.
+- **Privacy and Security**: Supports selective disclosure (e.g., proving age without revealing birthdate) and offline verification.
+- **Compliance**: Aligned with regulations like eIDAS 2.0.
+- **Efficiency**: Leverages OIDC mechanisms to streamline issuance while maintaining trust in the "triangle of trust" model (Issuer-Holder-Verifier).
+
+=== Verifiable Credential Formats
+
+OpenID4VCI supports multiple credential formats, which define how verifiable credentials are encoded and exchanged. *Keycloak currently supports SD-JWT VC for issuance*, while other formats are part of the OpenID4VCI specification.
+
+- https://drafts.oauth.net/oauth-sd-jwt-vc/draft-ietf-oauth-sd-jwt-vc.html[SD-JWT VC] – Selective Disclosure JSON Web Token Verifiable Credential
+- https://identity.foundation/jwt-vc-presentation-profile/[JWT VC] – JSON Web Token Verifiable Credential
+- https://www.iso.org/standard/69084.html[mDL/mdoc] – Mobile driver’s license / mobile document format
+
+These formats allow issuers to provide portable, verifiable digital proofs, supporting selective disclosure and offline verification in decentralized identity ecosystems.
```

Comment:
I'd also move the "*Keycloak currently supports SD-JWT VC for issuance*, while other formats are part of the OpenID4VCI specification."-part after this sentense.